src/lib/prepareOas.ts

```diff
@@ -18,28 +27,94 @@ export default async function prepareOas(path: string, command: 'openapi' | 'val
-        }
-
-        info(
-          chalk.yellow(`We found ${file} and are attempting to ${command === 'openapi' ? 'upload' : 'validate'} it.`)
-        );
-        resolve(file);
+    /**
+     * Scans working directory for a potential OpenAPI or Swagger file.
+     * Any files in the `.git` directory or defined in a top-level `.gitignore` file
+     * are skipped.
+     *
+     * A "potential OpenAPI or Swagger file" is defined as a YAML or JSON file
+     * that has an `openapi` or `swagger` property defined at the top-level.
+     *
+     * If multiple potential files are found, the user must select a single file.
+     *
+     * An error is thrown in the following cases:
+     * - if in a CI environment and multiple files are found
+     * - no files are found
+     */
+
+    const fileFindingSpinner = ora({ text: 'Attempting to locate API definitions...', ...oraOptions() }).start();
+
+    const action = command === 'openapi' ? 'upload' : 'validate';
+
+    const ignoreFilter = ignore().add('.git/');
+
+    if (fs.existsSync('.gitignore')) {
+      debug('.gitignore file found, adding to ignore filter');
+      ignoreFilter.add(fs.readFileSync('.gitignore').toString());
+    }
+
+    const jsonAndYamlFiles = readdirRecursive('.', ignoreFilter.createFilter()).filter(
+      file =>
+        file.toLowerCase().endsWith('.json') ||
+        file.toLowerCase().endsWith('.yaml') ||
+        file.toLowerCase().endsWith('.yml')
+    );
+
+    debug(`number of JSON or YAML files found: ${jsonAndYamlFiles.length}`);
+
+    const possibleSpecFiles = (
+      await Promise.all(
+        jsonAndYamlFiles.map(async file => {
+          debug(`attempting to oas-normalize ${file}`);
+          const oas = new OASNormalize(file, { enablePaths: true });
+          return oas
+            .version()
+            .then(version => {
+              debug(`OpenAPI/Swagger version for ${file}: ${version}`);
+              return version ? file : '';
+            })
+            .catch(e => {
+              debug(`error extracting OpenAPI/Swagger version for ${file}: ${e.message}`);
+              return '';
+            });
+        })
+      )
+    ).filter(Boolean);
+
+    debug(`number of possible OpenAPI/Swagger files found: ${possibleSpecFiles.length}`);
+
+    if (possibleSpecFiles.length === 0) {
+      fileFindingSpinner.fail();
+      throw new Error(
+        `We couldn't find an OpenAPI or Swagger definition.\n\nPlease specify the path to your definition with \`rdme ${command} ./path/to/api/definition\`.`
+      );
+    }
+
+    specPath = possibleSpecFiles[0];
+
+    if (possibleSpecFiles.length === 1) {
+      fileFindingSpinner.succeed(`${fileFindingSpinner.text} found! 🔍`);
+      info(chalk.yellow(`We found ${specPath} and are attempting to ${action} it.`));
+    }
+
+    if (possibleSpecFiles.length > 1) {
+      /* istanbul ignore next */
+      if ((ciDetect() && process.env.NODE_ENV !== 'testing') || process.env.TEST_CI) {
```

Comment:
Do we still need to check `NODE_ENV` if we have the `TEST_CI` env var?

Reply:
Sadly yes, since `ciDetect()` returns true when running our tests in GitHub Actions and we want to only test this if-block in specific tests.